asistente_ladm_col/tests/test_toolbar.py

```diff
@@ -0,0 +1,125 @@
+                        'db813999-3e1e-4539-99b6-7b797b1a3a72', '9831e428-040b-48e1-960a-5546fa47b99e',
+                        '9a009adf-6e87-4b04-aeb6-5d0830a848a7', '25509907-7cbc-4c7d-ab76-153898205f80',
+                        '1ef74b53-46fa-4dc5-9d85-c25db94f3dcd', '5b7194f2-54d8-4352-a47d-fac7f270a4a1']
+
+        exp = "{} in ('{}')".format(db_gpkg.names.T_ILI_TID_F,
+                                    "', '".join(boundary_ids))
+        layers[db_gpkg.names.LC_BOUNDARY_T].selectByExpression(exp)
+
+        self.toolbar.build_boundary(db_gpkg)
+
+        layers_feature_count = {db_gpkg.names.LC_BOUNDARY_POINT_T: 12,
+                                db_gpkg.names.LC_BOUNDARY_T: 22,
+                                db_gpkg.names.LC_PLOT_T: 14,
+                                db_gpkg.names.MORE_BFS_T: 43,
+                                db_gpkg.names.LESS_BFS_T: 0,
+                                db_gpkg.names.POINT_BFS_T: 47}
+
+        for layer_name, feature_count in layers_feature_count.items():
+            self.assertEqual(layers[layer_name].featureCount(),
+                             feature_count,
+                             'Features count does not match for the layer {}'.format(layer_name))
+
+        print('\nINFO: Check build boundaries for all features...')
+        layers[db_gpkg.names.LC_BOUNDARY_T].selectAll()
+        self.toolbar.build_boundary(db_gpkg)
+
+        layers_feature_count = {db_gpkg.names.LC_BOUNDARY_POINT_T: 12,
+                                db_gpkg.names.LC_BOUNDARY_T: 25,
+                                db_gpkg.names.LC_PLOT_T: 14,
+                                db_gpkg.names.MORE_BFS_T: 44,
+                                db_gpkg.names.LESS_BFS_T: 0,
+                                db_gpkg.names.POINT_BFS_T: 50}
+
+        for layer_name, feature_count in layers_feature_count.items():
+            self.assertEqual(layers[layer_name].featureCount(),
+                             feature_count,
+                             'Features count does not match for the layer {}'.format(layer_name))
+
+    def test_build_boundaries_with_empty_geom(self):
+        print('\nINFO: Validating build boundaries with features with null geometry...')
+        db_gpkg = get_copy_gpkg_conn('test_toolbar_empty_geom_gpkg')
+        res, code, msg = db_gpkg.test_connection()
+        self.assertTrue(res, msg)
+
+        boundary_layer = self.app.core.get_layer(db_gpkg, db_gpkg.names.LC_BOUNDARY_T, load=True)
+        self.assertEqual(boundary_layer.featureCount(), 3, 'Features count does not match for the layer boundary layer')
+
+        exp = "is_empty_or_null($geometry)"
+        boundary_layer.selectByExpression(exp)
+        self.assertEqual(boundary_layer.selectedFeatureCount(), 1,
+                         'Null features count does not match for the layer boundary layer')
+
+        boundary_layer.selectAll()
+        self.toolbar.build_boundary(db_gpkg)
```

Comment:
The method build_boundaries should be called like this:

    build_boundary(db_gpkg, False)  # where False is do not use selection

Reply:
Adjustment made, available in commit 513f31a